### ISSUE TITLE
Support JanusGraph 1.0.0

### ISSUE THREAD

**Issue by FlorianHockmann** [JanusGraph/janusgraph-dotnet#125]
This includes (at least) the following tasks:

- Update Gremlin.Net to 3.6
- Remove fallback for serialization of predicates for JanusGraph version < 0.6 (see [changelog entry](https://docs.janusgraph.org/master/changelog/#remove-support-for-old-serialization-format-of-janusgraph-predicates))

JanusGraph 1.0.0 also allows us to add support for Geoshapes with GraphBinary to JanusGraph.Net as that serialization now isn't specific to the Java implementation used for Geoshapes anymore (see [changelog](https://docs.janusgraph.org/master/changelog/#breaking-change-for-geoshape-graphbinary-serialization). But we can do that with a follow-up contribution.

I think it makes sense to create a branch here for the 0.4 version of JanusGraph.Net that is compatible with JanusGraph versions 0.6 and earlier in case we want to create another release for that in the future.

Last issue we have to decide on is the version of the release that will support JanusGraph 1.0.0. I think it makes sense to also release JanusGraph.Net with 1.0.0 for this because:
a) it signals that JanusGraph.Net is ready for production usage which it should be, considering that the version version has been released nearly 4 years ago and
b) we can align versions with JanusGraph versions which is probably easier to understand for users.

If there are any concerns with this approach, then please just comment here.

To summarize, the required tasks are:

- [x] Create a [`v0.4`](https://github.com/JanusGraph/janusgraph-dotnet/tree/v0.4) branch.
- [x] Update Gremlin.Net to 3.6: #126.
- [x] Remove fallback for outdated serialization of predicates: #126.
- [x] Bump version of JanusGraph.Net to 1.0.0 & update [version compatibility matrix](https://github.com/JanusGraph/janusgraph-dotnet#version-compatibility).
